src/accessControl/acp/index.js

```diff
@@ -385,14 +417,92 @@ export default class AcpAccessControlStrategy {
     this.#fetch = fetch;
   }
 
+  getInheritedPoliciesUrlsForResource() {
+    // get all policies that apply to this resource
+    const policies = acpv2.getPolicyUrlAll(this.#originalWithAcr);
+    // filter out owner and own resource policies
+    const namedPolicyUrls = namedPolicies.map(({ name }) => {
+      const containerUrl = getPolicyResourceUrl(
+        this.#originalWithAcr,
+        this.#policiesContainerUrl,
+        name
+      );
+      return getNamedPolicyUrl(containerUrl, name);
+    });
+    const inheritedPoliciesUrls = policies.filter(
+      (policyUrl) =>
+        !namedPolicyUrls.includes(policyUrl) && !policyUrl.includes("#Owner")
+    );
+    return inheritedPoliciesUrls;
+  }
+
+  async getInheritedPermissionsForResource() {
+    const policiesUrls = this.getInheritedPoliciesUrlsForResource();
+    if (!policiesUrls.length) return [];
+    try {
+      const modesAndAgents = await Promise.all(
+        policiesUrls.map(async (url) => {
+          const dataset = await getSolidDataset(url, {
+            fetch: this.#fetch,
+          });
+          const modesAndAgentsForUrl = getNamedPolicyModesAndAgents(
+            url,
+            dataset
+          );
+          return modesAndAgentsForUrl;
+        })
+      );
+      const permissions = [];
```

Comment:
You declare `permissions` as an array, but treat it as an object (map) later on. Let's change this to `const permissions = {};`